docs/index.bs

```diff
@@ -3481,93 +3452,79 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
+                  1. If the result of running [=Query Cache=] with |operation|.{{CacheBatchOperation/request}}, |operation|.{{CacheBatchOperation/options}}, and |addedItems| [=list/is not empty=], [=throw=] an "{{InvalidStateError}}" {{DOMException}}.
+                  1. Let |requestResponseList| be the result of running [=Query Cache=] with |operation|.{{CacheBatchOperation/request}} and |operation|.{{CacheBatchOperation/options}}.
+                  1. [=list/For each=] |item| in |requestResponseList|:
+                      1. If |operation|.{{CacheBatchOperation/type}} matches "delete", [=list/remove=] |item| from |cache|.
                   1. If |operation|.{{CacheBatchOperation/type}} matches "put", then:
                       1. If |operation|.{{CacheBatchOperation/response}} is null, <a>throw</a> a <code>TypeError</code>.
                       1. Let |r| be |operation|.{{CacheBatchOperation/request}}'s associated [=Request/request=].
                       1. If |r|'s [=request/url=]'s [=url/scheme=] is not one of "<code>http</code>" and "<code>https</code>", <a>throw</a> a <code>TypeError</code>.
                       1. If |r|'s [=request/method=] is not \`<code>GET</code>\`, <a>throw</a> a <code>TypeError</code>.
                       1. If |operation|.{{CacheBatchOperation/options}} is not null, <a>throw</a> a <code>TypeError</code>.
-                      1. Set |requestResponseArray| to the result of running <a>Query Cache</a> algorithm passing |operation|.{{CacheBatchOperation/request}}.
-                      1. If |requestResponseArray| is not an empty array, then:
-                          1. Let |requestResponse| be |requestResponseArray|[0].
-                          1. Let |fetchingRecord| be the corresponding <a>fetching record</a> for |requestResponse|[0] and |requestResponse|[1] in <a>request to response map</a>.
-                          1. Set |fetchingRecord|.\[[key]] to |operation|.{{CacheBatchOperation/request}} and |fetchingRecord|.\[[value]] to |operation|.{{CacheBatchOperation/response}}.
-                      1. Else:
-                          1. Set a newly-created <a>fetching record</a> {\[[key]]: |operation|.{{CacheBatchOperation/request}}, \[[value]]: |operation|.{{CacheBatchOperation/response}}} to <a>request to response map</a>.
-
-                          Note: The cache commit is allowed as long as the response's headers are available.
-
+                      1. Set |requestResponseList| to the result of running [=Query Cache=] with |operation|.{{CacheBatchOperation/request}}.
+                      1. If |requestResponseList| [=list/is not empty=], [=list/replace=] the [=list/item=] of |cache| that matches |requestResponseList|[0] with |operation|.{{CacheBatchOperation/request}}/|operation|.{{CacheBatchOperation/response}}.
```

Comment:
What if there are multiple matches, shouldn't we be removing those? It might be easiest to remove all matches then just append the new entry.

Reply:
Oh this is now done in addAll, but I'm not sure if it's right.

Reply:
I addressed it as you suggested. I think we still can do better for the return value of Batch Cache Operations somehow. I'll look into it as a separate work.